tango/common/testing.py

```diff
@@ -145,3 +172,146 @@ def requires_gpus(test_method):
             test_method
         )
     )
+
```

Comment:
Can you move these to a separate module? I don't want these steps to necessarily be registered when someone just wants to import the `TangoTestCase` class.

I would suggest making `tango/common/testing/` directory, and move this file to `tango/common/testing/__init__.py`, then have a separate file `tango/common/testing/steps.py` where you put these steps (don't import anything from `steps.py` into `__init__.py`).

Reply:
Thanks for the review - no worries, I was slow too!

> I don't want these steps to necessarily be registered when someone just wants to import the `TangoTestCase` class.

I don't see how that would happen except with `*` imports, which are discouraged anyway?

I mean I can do the change of course, I just don't really understand the gain...

Reply:
If those test steps are in the same file as `TangoTestCase` they will be registered when you import `TangoTestCase` even if you don't import those steps since every global statement in the file, including the `@Step.register()` statements, will be evaluated on import.

You can check this:

```python
from tango.step import Step

assert Step.list_available() == []

from tango.common.testing import TangoTestCase

assert "float" in Step.list_available()
```

Reply:
Actually, there is one substantial advantage of this approach - the diff gets much smaller. ;-)

Reply:
> If those test steps are in the same file as `TangoTestCase` they will be registered when you import `TangoTestCase` even if you don't import those steps since every global statement in the file, including the `@Step.register()` statements, will be evaluated on import.

Ah, I was not aware there were some global shenanigans going on 😅 

Also, it seems our responses were seconds apart. ;-)